### PR TITLE
Biodome vault, and a bit of poking at bots

### PR DESCRIPTION
**Pull request by ghost**
Biodome

Bit of a reference to Silent Running, and to lighten the mood from all the Zombie Jabronis, minigames, interstellar shopping malls, etc.

Should have everything needed to be able to do basic botany, solars should be linked properly as of last testing, and comes with some fossilized seeds, a chem dispenser, mortar and pestle, and a unique farmbot. Generally just a place to take refuge from the round and chill with some comfy farming.

Sprites for said farmbot courtesy of Niggly

ADDITIONAL - Tweaked the bots overall so that you can actually make custom versions of them (Seriously, why hardcode icon states?), as well as made it so that, depending on the medkit, certain medbots can heal certain things better.
#8823

### CODE REVIEW

**Comment by ghost** [comment]
Ohboy where did those 3 other commits come from

At least they seem to have nulled each other out

**Comment by 9600bauds** [comment]
Is it the biodome, or the bioctagon?
basic_circular_tower_layouts

File: code/game/machinery/bots/medbot.dm [hunk 1]
```diff
@@ -70,21 +71,29 @@
...
+					treatment_oxy = "dexalin"
 		else
-			src.botcard.access = src.botcard_access
+			treatment_brute = "bicaridine"
+	src.botcard = new /obj/item/weapon/card/id(src)
+	if(isnull(src.botcard_access) || (src.botcard_access.len < 1))
+		var/datum/job/doctor/J = new/datum/job/doctor
+		src.botcard.access = J.get_access()
+	else
+		src.botcard.access = src.botcard_access
```
Comment (Intigracy) [inline code/game/machinery/bots/medbot.dm]: Why'd you move this block out from under the spawn(4)?

**Comment by ihadtoregisterforthis** [comment]
Domestation when?

**Comment by AutismSpider** [comment]
@ihadtoregisterforthis call donut station 

donutmuthafucckars

**Comment by Probe1** [comment]
> Genetics is as big as Medbay

I was thinking about donutstation just yesterday when people asked why I don't make a map.  It's such a fun design but I think it's really shit to play in reality.  It'd need a hard gimmick like teleporters or mass drivers or transit tubes.

File: code/game/machinery/bots/medbot.dm (hunk 1 above)
Comment (ghost) [inline code/game/machinery/bots/medbot.dm]: Good spot, didn't realize it had moved

**Comment by clusterfack** [comment]
@Probe1 You'd be wrong

**Comment by Probe1** [comment]
About which part?

**Comment by FudgePucker** [comment]
You know what?
We should port some of the goonstation items that are unique and just have them as random vault loot.

You could have the plant lamp thats in that one derelict botany research ship in the asteroid field.

Or a fancy botanist jumpsuit.

Goon has a TON of unique items just fucking laying around in their space areas.

**Comment by ghost** [comment]
Would it be worth making a changelog regarding the fact that the chosen medkit for medbots matters?

**Comment by Intigracy** [comment]
Yes.

**Comment by ghost** [comment]
So, what else should I do with this?

File: maps/randomvaults/objects.dm [hunk 2]
```diff
@@ -136,3 +139,20 @@
 /obj/item/weapon/paper/iou
 	name = "paper- 'IOU'"
 	info = "I owe you a rod of destruction. Redeemable at Milliway's at the end of time."
+
+/obj/machinery/floodlight/on
+
+	New()
```
Comment (ComicIronic) [inline maps/randomvaults/objects.dm:145]: YOU DID THIS

Comment (ghost) [inline maps/randomvaults/objects.dm:145]: OH GOD, I CAN EXPLAIN

SEE THE VOX CHICKEN THING, I DID THE SAME MISTAKE

"I asked in IRC, and all I got was memes"

Comment (9600bauds) [inline maps/randomvaults/objects.dm:145]: Memes Kill

Comment (ghost) [inline maps/randomvaults/objects.dm:145]: So wait, what did this do?

This isn't the infinite feedback loop thing that I've heard about, is it?

Comment (ComicIronic) [inline maps/randomvaults/objects.dm:145]: No it's just shit code to do relative pathing and I can't believe it's not clusterfack catching this

**Pull request merged**